test/parser-test.ts

```diff
@@ -121,6 +121,31 @@ describe('DICOM parser', () => {
         });
     });
 
+    it('should not inflate deflated dataset until all dataset bytes have been read', () => {
+        const bytes = concatv(
+            data.fmiGroupLength(data.transferSyntaxUID(UID.ExplicitVRLittleEndian)),
+            data.transferSyntaxUID(UID.ExplicitVRLittleEndian),
+            data.patientNameJohnDoe(),
+            data.studyDate()
+        );
+        const deflatedBytes = concatv(
+            data.fmiGroupLength(data.transferSyntaxUID(UID.DeflatedExplicitVRLittleEndian)),
+            data.transferSyntaxUID(UID.DeflatedExplicitVRLittleEndian),
+            util.deflate(concatv(data.patientNameJohnDoe(), data.studyDate()))
+        );
+        const stop = (attributeInfo: AttributeInfo, depth: number): boolean =>
+            depth === 0 && 'tag' in attributeInfo && (attributeInfo as any).tag >= Tag.PatientName;
+        const testParser = new Parser(stop);
+        testParser.parse(bytes.subarray(0, bytes.length - 1));
+        assert(testParser.isComplete());
```

Comment:
I don't understand this assertion, the parser is fed almost all the bytes bar the last one, and yet it is complete?

Reply:
This was just to assert that the parser is complete when the stop attribute has been reached. This is also tested in anther test so I will remove it here.